### PR TITLE
[WIP] Port communicator.cpp to python

### PR DESCRIPTION
**Pull request by rht**
@maartenscholl  I am not so sure about this implementation. Need review.

### CODE REVIEW

File: esl/interaction/communicator.py [hunk 1]
```diff
@@ -1,5 +1,65 @@
...
\ No newline at end of file
+    def __init__(self, schedule, locked: bool = False):
+        self.locked: bool = locked
+        self.schedule = schedule
+
+    def process_message(self, message, step, seed) -> TimePoint:
+        first_event_ = step.upper
+
+        callback_ = self.callbacks_.find(message.type)
+        if self.callbacks_[-1] == callback_:
```
[the remark this answers is not in the document]
Comment (rht) [reply esl/interaction/communicator.py]: @maartenscholl fix applied in the second commit.

**Comment by rht** [comment]
@maartenscholl pushed into the 3rd commit 'tobesquashed2'.

File: esl/interaction/communicator.py [hunk 2]
```diff
@@ -1,5 +1,63 @@
...
+            for p, cf in callback_.second.items():
+                highest_priority = max(highest_priority, p)
+
+            priority_[highest_priority] = priority_.get(highest_priority, list()) + [m]
+
+        first_event_ = step.upper
+
+
+        for k, equal_ in reversed(priority_.items()):
+            if Scheduling.RANDOM == self.schedule:
```
Comment (rht) [inline esl/interaction/communicator.py]: @maartenscholl my linter says syntax error here.

Comment (rht) [reply esl/interaction/communicator.py]: Fixed by defining Scheduling outside of the Communicator class.

**Comment by maartenscholl** [comment]
closing because of a move to native code

**Pull request closed without merging**